### ISSUE TITLE
`unicorn/no-for-loop` false positive: loop through a string (returned by `string.slice()`)

### ISSUE THREAD

**Issue by panmenghan** [sindresorhus/eslint-plugin-unicorn#1802]
```js
const str = '123'.slice(1) // <-- maybe string.slice() triggers the bug

for (let i = 0; i < str.length; i++) { // <-- unicorn/no-for-loop
  console.log(str[i], i)
}
```

after auto-fix:
```js
const str = '123'.slice(1)

for (const [i, element] of str.entries()) { // <-- Property 'entries' does not exist on type 'string'
  console.log(element, i)
}
```
